### PR TITLE
[ASCII-873] bump go version to 1.20.11

### PR DESCRIPTION
**Pull request by GustavoCaso**
### What does this PR do?
Upgrade go version from 1.20.10 to 1.20.11
* [release notes](https://go.dev/doc/devel/release#go1.20.minor)

### Describe how to test/QA your changes
Sanity check should be covered by the CI and QA deploy process.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: tasks/update_go.py [hunk 1]
```diff
@@ -23,6 +23,7 @@ def go_version(_):
         "image_tag": "Tag from buildimages with format v<build_id>_<commit_id>",
         "test_version": "Whether the image is a test image or not",
         "warn": "Don't exit in case of matching error, just warn.",
+        "release_note": "Whether to create a releaes or not. The default behaviour is to create a releaes note"
```
Comment (pgimalac) [inline tasks/update_go.py:26]:
```suggestion
        "release_note": "Whether to create a release note or not. The default behaviour is to create a release note"
```

**Review by pgimalac — APPROVED** [review]

**Review by buraizu — APPROVED** [review]

File: .circleci/config.yml [hunk 2]
```diff
@@ -15,7 +15,7 @@ experimental:
 templates:
   job_template: &job_template
     docker:
-      - image: datadog/agent-buildimages-circleci-runner:v22284534-4d24e62
+      - image: datadog/agent-buildimages-circleci-runner_test_only:v23028262-7cab0fac
```
Comment (chouquette) [inline .circleci/config.yml:18]: For the record, beware when fixing the conflict here, the images are now pulled from GCR

**Review by chouquette — APPROVED** [review]

File: tasks/pipeline.py [hunk 3]
```diff
@@ -891,7 +891,7 @@ def update_circleci_config(file_path, image_tag, test_version):
     """
     Override variables in .gitlab-ci.yml file
     """
-    image_name = "datadog/agent-buildimages-circleci-runner"
+    image_name = "gcr.io/datadoghq/agent-circleci-runner"
```
Comment (GustavoCaso) [inline tasks/pipeline.py:894]: This change is because we changed the registry we pull images from https://github.com/DataDog/datadog-agent/pull/20596

Comment (chouquette) [reply tasks/pipeline.py:894]: nice catch!

**Review by chouquette — APPROVED** [review]

**Review by Kaderinho — APPROVED** [review]

**Pull request merged**